### PR TITLE
FEATURE: Add extra checks to ping_jump test

### PR DESCRIPTION
**Pull request by pmnoxx**
Due to recent changes bin `near-networking`, where we moves Routing Table calculation to another thread. This action is no longer synchronous.

I added extra checks to make sure that the routing table is correct.

I saw in the output logs that the routing failed. I want to figure out whenever that was due to routing table not being updated properly on all nodes, or for some other reason.

I caused failing CI, and therefore we were unable to do a merge.
For example: https://buildkite.com/nearprotocol/nearcore/builds/6774#c2582722-b8ab-49e0-8ee1-0c98fdca9b85

### CODE REVIEW

**Comment by pmnoxx** [comment]
@bowenwang1996 I added extra checks to make sure there are not issues with routing tables. If that happens again we will know.

File: integration-tests/tests/network/routing.rs [hunk 1]
```diff
@@ -103,6 +103,8 @@ fn ping_jump() {
     runner.push(Action::AddEdge(0, 1));
     runner.push(Action::AddEdge(1, 2));
     runner.push(Action::CheckRoutingTable(0, vec![(1, vec![1]), (2, vec![1])]));
+    runner.push(Action::CheckRoutingTable(1, vec![(0, vec![0]), (2, vec![2])]));
+    runner.push(Action::CheckRoutingTable(2, vec![(0, vec![1]), (1, vec![1])]));
```
Comment (bowenwang1996) [inline integration-tests/tests/network/routing.rs:107]: What do these checks do?

Comment (pmnoxx) [reply integration-tests/tests/network/routing.rs:107]: @bowenwang1996 In this test we start 3 nodes. We were checking the routing table only one one 0. We should check it on all nodes 0,1,2.

Comment (pmnoxx) [reply integration-tests/tests/network/routing.rs:107]: @bowenwang1996 If anything happens, we will know whenever there is an issue with routing table or with ping pong.

**Comment by pmnoxx** [comment]
@bowenwang1996 I'm trying to reproduce it locally. However, having extra checks is a good idea for the future.

**Comment by pmnoxx** [comment]
@bowenwang1996 I updater the PR description to explain the reason.

File: integration-tests/tests/network/routing.rs [hunk 2]
```diff
@@ -103,6 +103,8 @@ fn ping_jump() {
     runner.push(Action::AddEdge(0, 1));
     runner.push(Action::AddEdge(1, 2));
     runner.push(Action::CheckRoutingTable(0, vec![(1, vec![1]), (2, vec![1])]));
+    runner.push(Action::CheckRoutingTable(1, vec![(0, vec![0]), (2, vec![2])]));
```
Comment (mm-near) [inline integration-tests/tests/network/routing.rs:106]: comment about line 100:

can you add the comment to the test, describing what it is testing

**Review by mm-near — APPROVED** [review]

File: integration-tests/tests/network/routing.rs (hunk 2 above)
Comment (pmnoxx) [reply integration-tests/tests/network/routing.rs:109]: I added comments.

**Pull request merged**